### PR TITLE
#188 Add clean_tags option in harvester

### LINKED ISSUE #169: Tag max length to 50 instead of 100
Harvesting keywords, Tags names are truncated to 50 characters:

        tags = []
        if 'tags' in iso_values:
            for tag in iso_values['tags']:
                tag = tag[:50] if len(tag) > 50 else tag
                tags.append({'name': tag})

In DB Tag table there is a limit of 100 characters instead.
Is there a reason why Tag names are limited to 50 characters ?

### LINKED ISSUE #188: Spatial harvesters do not provide an option for tag cleanup
Harvested documents may contain keywords with unacceptable chars, that will be imported as tags without any chance to be cleaned.

An option similar to the one provided by the CKAN harvester should be provieded, i.e.:

    {"clean_tags": true}

### LINKED ISSUE #190: Missing doc for harvester configuration
Missing documentation for the per-harvester JSON config.

### PR DESCRIPTION
**Pull request by etj**
Fix #188 
Fix #190 
Fix #169

### CODE REVIEW

**Comment by etj** [comment]
(PR to be squashed, of course :) . I left all the commits in to preserve the original author).

**Comment by etj** [comment]
Still working on fixing tests...

**Comment by etj** [comment]
@amercader this PR is on the same line of ckan/ckanext-harvest#303 and ckan/ckanext-dcat#104 but for CSW harvesters, so I guess this PR can be useful as well. Any feedback? TIA

**Comment by tdipisa** [comment]
Hello @amercader, do you have some feedbacks about this PR ?

**Comment by tdipisa** [comment]
Dear @amercader, do you have some news about this PR? Are there some possibilities to merge it?

**Comment by amercader** [comment]
The changes look good, the test is a bit hard to follow, perhaps just adapt one of the existing ones to check the tags? Also I recommend that you look into the tests factories provided in CKAN core to make easier to create datasets, users, etc (`ckan.tests.factories`)

File: ckanext/spatial/tests/test_harvest.py [hunk 1]
```diff
@@ -801,6 +816,116 @@ def test_harvest_import_command(self):
...
+        if org is None:
+            org  = call_action('organization_create',
+                                context={'user': user_name},
+                                name='test-org')
+        existing_g = Group.by_name('existing-group')
+        if existing_g is None:
+            existing_g  = call_action('group_create',
+                                      context={'user': user_name},
+                                      name='existing-group')
+
```
[the remark this answers is not in the document]
Comment (cezio) [reply ckanext/spatial/tests/test_harvest.py]: AFAIR, i had problems with session management when running single test, so I added that code. It is not needed when running full test suite though. I'll remove it.

File: ckanext/spatial/tests/test_harvest.py [hunk 2]
```diff
@@ -52,16 +61,22 @@ def _create_job(self,source_id):
...
                  'user':u'harvest'}
+        if context_defaults:
+            context.update(context_defaults)
 
         if config.get('ckan.harvest.auth.profile') == u'publisher' \
            and not 'publisher_id' in source_fixture:
            source_fixture['publisher_id'] = self.publisher.id
 
         source_dict=get_action('harvest_source_create')(context,source_fixture)
+        if context.get('defer_commit'):
```
[the remark this answers is not in the document]
Comment (cezio) [reply ckanext/spatial/tests/test_harvest.py]: This is offspring of my problems with single test run. It will be removed.

**Comment by etj** [comment]
Hi @amercader, the fixes made according to your review have been merged in this PR.

**Comment by amercader** [comment]
@etj @tdipisa can you merge master? The tests are fixed there so we should know reliably if this ones are passing. Thanks

**Comment by tdipisa** [comment]
Hello @amercader, done :)
